yetanotherco/aligned_layer
# feat: first version of payment system

This PR implements the first version of payment system. The service manager contract now has a mapping `batchersBalances`, with balances for each batcher. 
`batcherAddress` is added to the BatchState struct of the service manager.

When the function call to respondToTask is called by the aggregator, the cost of the transaction is discounted from the batcher balance in the service manager and sent to the aggregator.

### Testing
The happy path can be tested checking the full flow with the usual commands.

For testing the new details from this PR
* Start anvil, aggregator and the operator as usual.
* Start batcher with
```
cargo +nightly-2024-04-17 run --manifest-path ./batcher/aligned-batcher/Cargo.toml --release -- --config ./config-files/config.yaml --env-file ./batcher/aligned-batcher/.env
```
* Send some proofs and check that the batch verification task transaction is reverted because batcher has no funds.
* Send some funds to the batcher with
```
make batcher_fund_service_manager_balance
```
* Send proofs and check that everything works as expected.

The commands to fund the batcher is the one that is run in `make batcher_start`.
